server/instance.go

```diff
@@ -552,26 +551,8 @@ func (s *Server) instanceCountGuard(ctx context.Context) error {
 // disallowBytebaseStore prevents users adding Bytebase's own Postgres database.
 // Otherwise, users can take control of the database which is a security issue.
 func (s *Server) disallowBytebaseStore(engine db.Type, host, port string) error {
-	if engine != db.Postgres {
-		return nil
-	}
-	if port != fmt.Sprintf("%v", s.datastorePort) {
-		return nil
-	}
-
-	existErr := fmt.Errorf("instance doesn't exist for host %q and port %q", host, port)
-	if host == "localhost" || host == "127.0.0.1" || host == "::1" {
-		return existErr
-	}
-	// Check loopback addresses
-	ips, err := net.LookupIP(host)
-	if err != nil {
-		return nil
-	}
-	for _, ip := range ips {
-		if ip.IsLoopback() {
-			return existErr
-		}
+	if engine == db.Postgres && port == fmt.Sprintf("%v", s.datastorePort) && host == common.GetPostgresDataDir(s.dataDir) {
```

Comment:
I might not fully get the "port == fmt.Sprintf("%v", s.datastorePort) && host == common.GetPostgresDataDir(s.dataDir)" logic wrt the unix domain socket change.

Reply:
This function disallowBytebaseStore means to reject the user creating an instance of Bytebase's Postgres instance. It was rejecting localhost:port connection, but there was still a bug that users'rel able to connect via /tmp/xxx socket file.

This change makes our Postgres listening on Unix socket only without listening on localhost:port (flag -h ""). And the change to function disallowBytebaseStore prevents the case above.

Reply:
If we only allow our Postgres store to listen on a local socket, how could the user specify a host:port combination from bytebase instance setting to connect to our store?

Reply:
Not any more, hence removing code below, but we are still doing a unix socket file path check.

port == fmt.Sprintf("%v", s.datastorePort) && host == common.GetPostgresDataDir(s.dataDir)

Reply:
Done.